### PR TITLE
WRR-626: Updated puppeteer and wait-on dependencies to latest version

### PR DESCRIPTION
**Pull request by daniel-stoian-lgp**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] I have run automated testing and it is passed
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Puppeteer needed upgrade to the latest version
wait-on needed upgrade to the latest version

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
changelog for puppeteer-core: https://github.com/puppeteer/puppeteer/blob/main/packages/puppeteer-core/CHANGELOG.md 
Updated puppeteer dependencies to latest version
"ws" subdependency had to be pinned to an older version "7.5.10" in order to fix security vulnerabilities

Fot wait-on, there were no breaking changes affecting us https://github.com/jeffbski/wait-on/releases 

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)
`executablePath` is required to be added since puppteer19 
`headless:true` was required because of this breaking change https://github.com/puppeteer/puppeteer/pull/11815 
jsdon-extended.js file was added because of a conflict between jsdom and new ReadableStreams introduced in puppeteer 22 https://github.com/puppeteer/puppeteer/pull/11805 . Solution could be found here: https://github.com/mswjs/msw/issues/1916#issuecomment-1850757977  

We had to add "ws": "7.5.10"in the overrides section of package.json in order to fix security vulnerabilities

### Links
[//]: # (Related issues, references)
WRR-626

### Comments

Enact-DCO-1.0-Signed-off-by: Daniel Stoian (daniel.stoian@lgepartner.com)

### CODE REVIEW

File: jsdom-extended.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+		this.global.Request = Request;
+		this.global.Response = Response;
+		this.global.Request = Request;
+		this.global.Response = Response;
+		this.global.fetch = fetch;
+		this.global.structuredClone = structuredClone;
+	}
+}
+
+module.exports = JSDOMEnvironmentExtended;
```
Comment (adrian-cocoara-lgp) [inline jsdom-extended.js:24]: add an empty line

Comment (daniel-stoian-lgp) [inline jsdom-extended.js:24]:
```suggestion
module.exports = JSDOMEnvironmentExtended;

```

**Review by adrian-cocoara-lgp — APPROVED** [review]
LGTM 👍

**Comment by hong6316** [comment]
Please add a comment about overriding ws v7.5.10. there's no description,

**Review by hong6316 — APPROVED** [review]
👍

**Pull request merged**